test/tests/functional/pbs_sched_preempt_enforce_resumption.py

```diff
@@ -0,0 +1,111 @@
+# trademark licensing policies.
+
+import time
+from tests.functional import *
+
+
+class TestSchedPreemptEnforceResumption(TestFunctional):
+    """
+    Test sched_preempt_enforce_resumption working
+    """
+    def setUp(self):
+        TestFunctional.setUp(self)
+        # Set default resources available on the default mom
+        a = {ATTR_rescavail + '.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        # Create an express queue
+        b = {ATTR_qtype: 'Execution', ATTR_enable: 'True',
+             ATTR_start: 'True', ATTR_p: '151'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, b, "expressq")
+        a = {ATTR_sched_preempt_enforce_resumption: True}
+        self.server.manager(MGR_CMD_SET, SCHED, a)
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+    def test_filler_job(self):
+        """
+        This test confirms that the filler job does not run if it conflicts
+        with running of a suspended job.
+        """
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 80})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_q: 'expressq',
+                           ATTR_l + '.walltime': 30})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_l + '.walltime': 90})
+        jid3 = self.server.submit(j3)
+        logmsg = ";Job would conflict with reservation or top job"
+        self.scheduler.log_match(jid3 + logmsg)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid3)
+
+    def test_suspended_job_ded_time_calendared(self):
+        """
+        This test confirms that a suspended job becomes top job when unable to
+        resume due to conflict with dedicated time.
+        """
+        now = time.time()
+        self.scheduler.add_dedicated_time(start=now + 180, end=now + 300)
+
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 120})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_l + '.walltime': 60,
+                           ATTR_q: 'expressq'})
+        j2.set_sleep_time(60)
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+
+        self.server.tracejob_match(
```

Comment:
Well, in this case, you are not really checking for conflicts with dedicated time, are you? I think it is better to check for estimated start time of the suspended job if you intend to test its interaction with dedicated time.

Reply:
We need not, as the issue at hand is that the suspended job never gets the estimated start time. But, it is good to test if the estimated start time is as per our expectations, so I have made the changes.

Reply:
Oh ok, I thought that was your intention in this test because the test above otherwise already captures that the suspended job made it into the calendar.